mava/wrappers/shared.py

```diff
@@ -120,11 +120,14 @@ def observation_spec(
         agents_view = specs.Array(
             (self._env.num_agents, num_obs_features), jnp.int32, "agents_view"
         )
+
+        if hasattr(obs_spec, "global_state"):
+            state_size = obs_spec.global_state.shape[-1] + self._env.num_agents
+        else:
+            state_size = num_obs_features * self._env.num_agents + self._env.num_agents
```

Comment:
I think we can collapse this condition into `self.has_global_state` we must just make sure has_global_state is always true for `JaxMarlEnvs`. And then also use your way of calculating state size (with the `obs_spec`) as it's more future proof.


```python
if self.has_global_state:
    wrapped_state_shape = obs_spec.global_state.shape
    state_shape = (*wrapped_state_shape[:-1], wrapped_state_shape[-1] + self._env.num_agents)
    global_state = specs.Array(state_shape, ...)
    return obs_spec.replace(agents_view=agents_view, global_state=global_state)

return obs_spec.replace(agents_view=agents_view)
```